WordPress/wordpress-playground
# Fixing build regression [BISON COMPILE]

## What is this PR doing?

This installs the necessary version of Bison in a separate docker image and uses that image as a base image for the PHP build. This allows us to only build Bison twice: once for v2.7 and again for v3.

## What problem is it solving?

Currently Bison 2.7 isn't installing correctly and thus we cannot build PHP <=7.3.

https://github.com/WordPress/wordpress-playground/blob/a4c3ad0e49143ec5893b0e11b0026157e2fc4951/packages/php-wasm/compile/php/Dockerfile#L4-L5

## How is the problem addressed?

* Installs the necessary version of Bison in a separate docker image and uses that image as a base image for the PHP build.

## Testing Instructions

Checkout the branch & install the dependencies:

```bash
git checkout sm-fixing-build-regression
npm install
```

Run the build and ensure it succeeds:

```bash
npm run recompile:php:node
npm run recompile:php:web
```

Start the dev server:

```bash
npm run dev
```

Then, navigate to:

* http://localhost:5400/website-server/?php=8.3
* http://localhost:5400/website-server/?php=8.2
* http://localhost:5400/website-server/?php=8.1
* http://localhost:5400/website-server/?php=8.0
* http://localhost:5400/website-server/?php=7.4
* http://localhost:5400/website-server/?php=7.3
* http://localhost:5400/website-server/?php=7.2
* http://localhost:5400/website-server/?php=7.1
* http://localhost:5400/website-server/?php=7.0

... and ensure that wordpress works as-expected in each environment.
